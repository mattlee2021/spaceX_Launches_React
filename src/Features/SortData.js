import styles from "./SortData.module.css";

const SortData = (props) => {
  const { setSortCriteria } = props;
  // similar to launchsitefilter
  // setLaunchSiteFilter getting called with the correct property;
  // snapshot -> snapshot test and logic testing
  // snapshot too since it is HTML

  return (
    <form>
      <label>Sort By: </label>
      <select
        onChange={(event) => {
          setSortCriteria(event.target.value);
        }}
        className={styles.sort_input}
      >
        <option value="">None</option>
        <option value="site_name_long">Launch Site</option>
        <option value="launch_date_local">Launch Date</option>
        <option value="mission_name">Mission Name</option>
      </select>
    </form>
  );
};

export default SortData;
