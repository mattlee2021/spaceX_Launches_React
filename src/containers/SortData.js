import {useState } from "react";

/**
 * This function sorts the displayed information from the SpaceX GraphQL API. Since the Launch Site and 
 * Mission Name fields are composed of words, this function will sort alphabetically according to the 
 * selected choice. For Launch Date, this function will sort the the launches in chronological order. 
 */

export function SortDataFeature(props){

    const[sort_state, setSort_state]= useState(""); 
    const {onSubmit}= props;

    const handleChange_Sort = (event) => {
        setSort_state(event.target.value);
      }
 
      const onSubmit_Sort = (event) => {
        event.preventDefault();
        onSubmit(sort_state);
      }
        
    return (
        <form onSubmit={onSubmit_Sort} className="Filtering">  
            <label htmlFor="sorting">Sort By: </label>
            <select onChange={handleChange_Sort} className="sort_box">
                    <option value="">None</option>
                    <option value="site_name_long">Launch Site</option>
                    <option value="launch_date_local">Launch Date</option>
                    <option value="mission_name">Mission Name</option>
            </select>
            <button type="submit">Submit</button>
        </form>
    );
}