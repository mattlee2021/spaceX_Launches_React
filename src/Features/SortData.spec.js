import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestRenderer from "react-test-renderer";
import SortData from "./SortData";

describe("SortData", () => {
  it("Should match the snapshot", () => {
    const testRenderer = TestRenderer.create(<SortData />);
    testRenderer.toJSON();
    expect(testRenderer).toMatchSnapshot();
  });

  it("Should have 4 options in dropdown", () => {
    const mockSetSortCriteria = jest.fn();
    render(<SortData setSortCriteria={mockSetSortCriteria} />);
    expect(screen.getAllByRole("option").length).toBe(4);
  });

  it("Should call setSortCriteria with site_name_long", () => {
    const mockSetSortCriteria = jest.fn();
    render(<SortData setSortCriteria={mockSetSortCriteria} />);
    userEvent.selectOptions(
      screen.getByRole("combobox"),
      screen.getByRole("option", { name: "Launch Site" })
    );
    expect(mockSetSortCriteria).toHaveBeenCalledWith("site_name_long");
  });

  it("Should call setSortCriteria with launch_date_local", () => {
    const mockSetSortCriteria = jest.fn();
    render(<SortData setSortCriteria={mockSetSortCriteria} />);
    userEvent.selectOptions(
      screen.getByRole("combobox"),
      screen.getByRole("option", { name: "Launch Date" })
    );
    expect(mockSetSortCriteria).toHaveBeenCalledWith("launch_date_local");
  });

  it("Should call setSortCriteria with mission_name", () => {
    const mockSetSortCriteria = jest.fn();
    render(<SortData setSortCriteria={mockSetSortCriteria} />);
    userEvent.selectOptions(
      screen.getByRole("combobox"),
      screen.getByRole("option", { name: "Mission Name" })
    );
    expect(mockSetSortCriteria).toHaveBeenCalledWith("mission_name");
  });
});
