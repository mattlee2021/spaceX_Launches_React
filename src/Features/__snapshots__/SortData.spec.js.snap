// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortData Should match the snapshot 1`] = `
<form>
  <label>
    Sort By: 
  </label>
  <select
    className="sort_input"
    onChange={[Function]}
  >
    <option
      value=""
    >
      None
    </option>
    <option
      value="site_name_long"
    >
      Launch Site
    </option>
    <option
      value="launch_date_local"
    >
      Launch Date
    </option>
    <option
      value="mission_name"
    >
      Mission Name
    </option>
  </select>
</form>
`;
